aoc-client/src/lib.rs

```diff
@@ -421,6 +421,76 @@ impl AocClient {
         Ok(())
     }
 
+    fn get_personal_stats_html(&self) -> AocResult<String> {
+        debug!("🦌 Fetching {} personal stats", self.year);
+
+        let url =
+            format!("https://adventofcode.com/{}/leaderboard/self", self.year);
+        let response = http_client(&self.session_cookie, "text/html")?
+            .get(url)
+            .send()?;
+
+        if response.status() == StatusCode::NOT_FOUND {
+            // A 402 reponse means the calendar for
+            // the requested year is not yet available
+            return Err(AocError::InvalidEventYear(self.year));
+        } else if response.status() == StatusCode::FOUND {
+            // A 302 reponse is a redirect and it likely
+            // means we're not logged in
+            warn!(
+                "🍪 It looks like you are not logged in, try logging in again"
+            );
+            return Err(AocError::AocResponseError);
+        }
+
+        let contents = response.error_for_status()?.text()?;
+
+        let main = Regex::new(r"(?i)(?s)<main>(?P<main>.*)</main>")
+            .unwrap()
+            .captures(&contents)
+            .ok_or(AocError::AocResponseError)?
+            .name("main")
+            .unwrap()
+            .as_str()
+            .to_string();
+
+        Ok(main)
+    }
+
+    pub fn show_personal_stats(&self) -> AocResult<()> {
+        let stats_html = self.get_personal_stats_html()?;
+        let stats_text = self.html2text(&stats_html);
+
+        // print explanatory paragraph before recoloring the text
+        for line in stats_text.lines().take_while(|line| !line.is_empty()) {
+            println!("{}", line);
+        }
+
+        let stats_text = stats_text
+            .replace(
+                "--------Part 1---------",
+                &"--------Part 1---------".color(SILVER).to_string(),
+            )
+            .replace(
+                "--------Part 2---------",
+                &"--------Part 2---------".color(GOLD).to_string(),
+            )
+            .replace("Time", &"Time".color(GOLD).to_string())
+            .replace("Rank", &"Rank".color(GOLD).to_string())
+            .replace("Score", &"Score".color(GOLD).to_string())
+            .replacen("Time", &"Time".color(SILVER).to_string(), 1)
+            .replacen("Rank", &"Rank".color(SILVER).to_string(), 2) // "Rank" also in explanation
+            .replacen("Score", &"Score".color(SILVER).to_string(), 2); // "Score" also in
```

Comment:
The words in the header are not being coloured as you printed them before (on lines 465-466). We can probably leave the explanation uncoloured though

Reply:
Sorry for being unclear in the comments, it was my intention for the explanation to remain uncolored since that is the behavior displayed on the website.